app/src/main/java/com/example/android/trackmysleepquality/sleeptracker/SleepTrackerViewModel.kt

```diff
@@ -82,12 +103,17 @@ class SleepTrackerViewModel(
     fun onClearTracking() {
         viewModelScope.launch {
             clear()
+            _showSnackbarEvent.value = true
         }
     }
 
     private suspend fun clear() {
         database.clear()
         tonight.value = null
     }
+
+    fun doneNavigating() {
+        _navigateToSleepQuality.value = null
```

Comment:
Em nên research về việc tại sao livedata lại emit lại giá trị cuối cùng, cách khắc phục nó, xài Event Wrapper nha
https://medium.com/androiddevelopers/livedata-with-snackbar-navigation-and-other-events-the-singleliveevent-case-ac2622673150

Reply:
Em đã sửa lại thành các Event Wrapper